In [ ]:
from google.colab import files

uploaded = files.upload()

Saving events_data.csv to events_data (3).csv


In [ ]:
print(uploaded['events_data.csv'][:1000].decode('utf-8') + '...')

event_id,device_id,timestamp,longitude,latitude,city,state
2765368,2973347786994914304,2016-05-07 22:52:05,77.22567599999998,28.730140000000002,Delhi,Delhi
2955066,4734221357723754978,2016-05-01 20:44:16,88.388361,22.660325,Calcutta,WestBengal
605968,-3264499652692492784,2016-05-02 14:23:04,77.25680899999998,28.757906,Delhi,Delhi
448114,5731369272434022440,2016-05-03 13:21:16,80.34361299999998,13.153332,Chennai,TamilNadu
665740,338888002570799963,2016-05-06 03:51:05,85.997745,23.842609,Bokaro,Jharkhand
1078723,-512424213979236423,2016-05-02 02:21:20,83.39824399999998,17.768148999999994,Visakhapatnam,AndhraPradesh
2948755,5536513450525271144,2016-05-03 16:20:14,80.359796,13.128174,Chennai,TamilNadu
280014,-8879643774076165575,2016-05-05 13:06:01,78.155397,16.390327000000006,Wanparti,AndhraPradesh
2309828,3020769016382465619,2016-05-04 09:24:50,77.263922,28.757914000000003,Delhi,Delhi
...


In [ ]:
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
import io

df = pd.read_csv(io.StringIO(uploaded['events_data.csv'].decode('utf-8')))

state_list = ['Rajasthan', 'UttarPradesh', 'Haryana', 'Jharkhand', 'Orissa', 'Telangana', 'Assam']

df

event_id     device_id  ...           city          state
0         2765368  2.973348e+18  ...          Delhi          Delhi
1         2955066  4.734221e+18  ...       Calcutta     WestBengal
2          605968 -3.264500e+18  ...          Delhi          Delhi
3          448114  5.731369e+18  ...        Chennai      TamilNadu
4          665740  3.388880e+17  ...         Bokaro      Jharkhand
...           ...           ...  ...            ...            ...
3252945   2687452 -1.937028e+18  ...           Pune    Maharashtra
3252946   1051580  3.345851e+18  ...         Mumbai    Maharashtra
3252947   1316227 -6.406040e+18  ...          Delhi          Delhi
3252948    381262 -2.920741e+18  ...  Visakhapatnam  AndhraPradesh
3252949    522592  3.212750e+18  ...         Kambam      TamilNadu

[3252950 rows x 7 columns]

Read events_data.csv. Total 3252950 rows.

dtype: object
event_id       int64
device_id    float64
timestamp     object
longitude    float64
latitude     float64
city          object
state         object

Rajasthan 1229 entries 
UttarPradesh 581 entries 
Haryana 116 entries 
Jharkhand 110 entries 
Orissa 100 entries 
Telangana 86 entries 
Assam 64 entries

In [ ]:
!pip install pymysql

     |████████████████████████████████| 43 kB 1.1 MB/s 


In [ ]:
import pymysql
from pymysql import FIELD_TYPE
mydb = pymysql.connect(
  host="cpanel.insaid.co",
  user="student",
  password="student",
  database="Capstone1",
  charset="utf8"
)

mycursor = mydb.cursor()

#mycursor.execute("SELECT Distinct Count(*) FROM gender_age_train")
mycursor.execute("SELECT * FROM gender_age_train LIMIT 10")

myresult = mycursor.fetchall()

for x in myresult:
  print(x)

mycursor.close()
mydb.close()

(-8076087639492063270, 'M', 35, 'M32-38')
(-2897161552818060146, 'M', 35, 'M32-38')
(-8260683887967679142, 'M', 35, 'M32-38')
(-4938849341048082022, 'M', 30, 'M29-31')
(245133531816851882, 'M', 30, 'M29-31')
(-1297074871525174196, 'F', 24, 'F24-26')
(236877999787307864, 'M', 36, 'M32-38')
(-8098239495777311881, 'M', 38, 'M32-38')
(176515041953473526, 'M', 33, 'M32-38')
(1596610250680140042, 'F', 36, 'F33-42')


Select gender_age_train dataset. Total Rows - 74645

In [ ]:
import pymysql
mydb = pymysql.connect(
  host="cpanel.insaid.co",
  user="student",
  password="student",
  database="Capstone1",
  charset="utf"
)

mycursor = mydb.cursor()

#mycursor.execute("SELECT Distinct Count(*) FROM phone_brand_device_model")
mycursor.execute("SELECT * FROM phone_brand_device_model LIMIT 10")

myresult = mycursor.fetchall()

for x in myresult:
  print(x)

mycursor.close()
mydb.close()

(1877775838486905855, 'vivo', 'Y13')
(-3766087376657242966, '小米', 'V183')
(-6238937574958215831, 'OPPO', 'R7s')
(8973197758510677470, '三星', 'A368t')
(-2015528097870762664, '小米', '红米Note2')
(-8896952452425303597, '小米', '红米Note3')
(-2634171391322875113, '小米', 'S5.1')
(-2526386806972133373, '三星', 'Galaxy Grand Prime')
(1427934937997164515, '小米', 'MI 2S')
(2866484947180243791, '三星', 'Galaxy Note 2')


Select phone_brand_device_model dataset. Total Rows - 87726

In [ ]:
import pymysql
mydb = pymysql.connect(
  host="cpanel.insaid.co",
  user="student",
  password="student",
  database="Capstone1",
  charset="utf8mb4"
)

mycursor = mydb.cursor()

mycursor.execute("SELECT Distinct * FROM gender_age_train gat INNER JOIN phone_brand_device_model pbdm ON gat.device_id = pbdm.device_id")

myresult = mycursor.fetchall()

for x in myresult:
  print(x)

mycursor.close()
mydb.close()

Streaming output truncated to the last 5000 lines.
(-4424632656786586870, 'M', 22, 'M22-', -4424632656786586870, '小米', '红米note')
(7310671919846805261, 'M', 33, 'M32-38', 7310671919846805261, '华为', 'Mate 7')
(-178127191997594818, 'M', 31, 'M29-31', -178127191997594818, '小米', '红米')
(-1646254635087266595, 'M', 24, 'M23-26', -1646254635087266595, '语信', '红辣椒')
(8518732759756746297, 'F', 28, 'F27-28', 8518732759756746297, '小米', 'MI 4')
(-3576581881172391657, 'M', 30, 'M29-31', -3576581881172391657, '华为', '荣耀6')
(7501344298120802275, 'M', 28, 'M27-28', 7501344298120802275, '联想 ', 'memo 5S')
(-8726733746650241932, 'M', 40, 'M39+', -8726733746650241932, '华为', '荣耀畅玩4X')
(4890287079456010222, 'M', 33, 'M32-38', 4890287079456010222, '华为', '荣耀畅玩4X')
(-3109135336578854167, 'M', 25, 'M23-26', -3109135336578854167, '魅族', '魅蓝NOTE')
(-3348672961789167113, 'F', 20, 'F23-', -3348672961789167113, '魅族', 'MX5')
(1759178146045384850, 'M', 22, 'M22-', 1759178146045384850, '三星', '乐檬K3 Note')
(430533724762329764

IOPub data rate exceeded.
The notebook server will temporarily stop sending output
to the client in order to avoid crashing it.
To change this limit, set the config variable
`--NotebookApp.iopub_data_rate_limit`.

Current values:
NotebookApp.iopub_data_rate_limit=1000000.0 (bytes/sec)
NotebookApp.rate_limit_window=3.0 (secs)



Streaming output truncated to the last 5000 lines.
(1551515816601496261, 'M', 30, 'M29-31', 1551515816601496261, '小米', '红米2')
(-7992099517730705745, 'M', 46, 'M39+', -7992099517730705745, 'vivo', 'Xperia C3')
(4567349784959138145, 'F', 37, 'F33-42', 4567349784959138145, '小米', '红米note')
(-5305190980379687886, 'F', 36, 'F33-42', -5305190980379687886, '三星', 'Galaxy Note 2')
(-8146516170823798740, 'F', 34, 'F33-42', -8146516170823798740, 'OPPO', 'R3')
(-229317349110458159, 'F', 28, 'F27-28', -229317349110458159, '三星', 'Galaxy Grand Neo Plus')
(3792232774004368858, 'M', 22, 'M22-', 3792232774004368858, 'vivo', 'S11T')
(-2740470823835504585, 'F', 26, 'F24-26', -2740470823835504585, 'OPPO', 'R7 Plus')
(-7327192879116136771, 'M', 35, 'M32-38', -7327192879116136771, '华为', 'HN3-U01')
(6284292811428649496, 'M', 27, 'M27-28', 6284292811428649496, '小米', 'ELIFE E6 mini')
(-3778991467807419572, 'M', 26, 'M23-26', -3778991467807419572, '一加', 'One')
(-6090702707636116922, 'F', 19, 'F23-', -609070270763

Join on gender_age_train and phone_brand_device_model on device_id. Total Rows - 74617.